test/reconnect_test.exs

```diff
@@ -0,0 +1,44 @@
+defmodule RigInboundGatewayWeb.ReconnectTest do
+  use ExUnit.Case, async: true
+
+  require Logger
+
+  alias HTTPoison
+  alias Rig.Connection.Codec
+
+  @event_hub_http_port Confex.fetch_env!(:rig, RigInboundGatewayWeb.Endpoint)[:http][:port]
+
+  test "Initialize connection, then disconnect and reconnect" do
```

Comment:
related to my other comment on test descriptions, this should describe what the test is actually testing. in this case, it seems to test that "Passing a connection token when connecting via SSE returns the passed connection token.". If that doesn't fit what you had in mind, you should probably think about the test's assertions and update the test description/name accordingly.